miio/g1vacuum.py

```diff
@@ -0,0 +1,362 @@
+    "home": {"siid": 2, "aiid": 3},
+    "find": {"siid": 6, "aiid": 1},
+    "start": {"siid": 2, "aiid": 1},
+    "stop": {"siid": 2, "aiid": 2},
+    "reset_main_brush_life_level":  {"siid": 14, "aiid": 1},
+    "reset_side_brush_life_level":  {"siid": 15, "aiid": 1},
+    "reset_filter_life_level":  {"siid": 11, "aiid": 1}
+    }
+}
+
+error_codes = {
+    0: "No error",
+    1: "Left Wheel stuck",
+    2: "Right Wheel stuck",
+    3: "Cliff error",
+    4: "Low battery",
+    5: "Bump error",
+    6: "Main Brush Error",
+    7: "Side Brush Error",
+    8: "Fan Motor Error",
+    9: "Dustbin Error",
+    10: "Charging Error",
+    11: "No Water Error",
+    12: "Pick Up Error"
+}
+
+class G1ChargeState(Enum):
+    Discharging = 0
+    Charging = 1
+    FullyCharged = 2
+
+class G1State(Enum):
+    """Vacuum Status"""
+    Idle = 1
+    Sweeping = 2
+    Paused = 3
+    Error = 4
+    Charging = 5
+    GoCharging = 6
+
+class G1Consumable(Enum):
+    """Consumables"""
+    MainBrush = "main_brush_life_level"
+    SideBrush = "side_brush_life_level"
+    Filter = "filter_life_level"
+
+class G1VacuumMode(Enum):
+    """Vacuum Mode"""
+    GlobalClean = 1
+    SpotClean = 2
+    Wiping = 3
+
+class G1WaterLevel(Enum):
+    """Water Flow Level"""
+    Level1 = 1
+    Level2 = 2
+    Level3 = 3
+
+class G1FanSpeed(Enum):
+    """Fan speeds, same as for ViomiVacuum."""
+    Mute = 0
+    Standard = 1
+    Medium = 2
+    High = 3
+
+class G1Languages(Enum):
+    Chinese = 0
+    English = 1
+
+class G1MopState(Enum):
+    Off = 0
+    On = 1
+
+
+class G1Status(DeviceStatus):
+    """Container for status reports from Mijia Vacuum G1."""
+    """
+        Response (MIoT format) of a Mijia Vacuum G1 (mijia.vacuum.v2)
+        [
+            {'did': 'battery', 'siid': 3, 'piid': 1, 'code': 0, 'value': 100}, 
+            {'did': 'charge_state', 'siid': 3, 'piid': 2, 'code': 0, 'value': 2}, 
+            {'did': 'error_code', 'siid': 2, 'piid': 2, 'code': 0, 'value': 0}, 
+            {'did': 'state', 'siid': 2, 'piid': 1, 'code': 0, 'value': 5}, 
+            {'did': 'fan_speed', 'siid': 2, 'piid': 6, 'code': 0, 'value': 1}, 
+            {'did': 'operating_mode', 'siid': 2, 'piid': 4, 'code': 0, 'value': 1}, 
+            {'did': 'mop_state', 'siid': 16, 'piid': 1, 'code': 0, 'value': 0}, 
+            {'did': 'water_level', 'siid': 2, 'piid': 5, 'code': 0, 'value': 2}, 
+            {'did': 'main_brush_life_level', 'siid': 14, 'piid': 1, 'code': 0, 'value': 99}, 
+            {'did': 'main_brush_time_left', 'siid': 14, 'piid': 2, 'code': 0, 'value': 17959},
+            {'did': 'side_brush_life_level', 'siid': 15, 'piid': 1}, 
+            {'did': 'side_brush_time_left', 'siid': 15, 'piid': 2}, 
+            {'did': 'filter_life_level', 'siid': 11, 'piid': 1}, 
+            {'did': 'filter_time_left', 'siid': 11, 'piid': 2},
+            {'did': 'clean_area', 'siid': 9, 'piid': 1}, 
+            {'did': 'clean_time', 'siid': 9, 'piid': 2}
```

Comment:
The payload seems to be incomplete, as these do not contain the code and the value? The idea behind having this in the code base is to allow someone unfamiliar without access to the device to gain some insight in case parts of the code requires changing, so it is sort of crucial to have this available.

Could you please recheck this?

Reply:
copy&paste gone wrong. Corrected.

I hope everythings OK now. Surely  I did learn a bit when working on this code, so no problem at all.